modules/javafx.graphics/src/main/java/javafx/scene/layout/BorderPane.java

```diff
@@ -626,8 +626,8 @@ private double getAreaWidth(Node child, double height, boolean minimum) {
     private double getAreaHeight(Node child, double width, boolean minimum) {
         if (child != null && child.isManaged()) {
             Insets margin = getNodeMargin(child);
-            return minimum ? computeChildMinAreaHeight(child, -1, margin, width, false):
-                                   computeChildPrefAreaHeight(child, -1, margin, width, false);
+            return minimum ? computeChildMinAreaHeight(child, -1, margin, width, true):
+                                   computeChildPrefAreaHeight(child, -1, margin, width, true);
```

Comment:
Note: before the last boolean parameter was introduced in the `computeChildXXXAreaHeight` functions in `Region`, the default behavior was to have fill width set to `true`.  The reason I picked `false` is because the calculation above in `getAreaWidth` was using `false`, but it looks now that this was never correct.